beacon-chain/core/validators/committees.go

```diff
@@ -102,3 +104,64 @@ func getCommitteesPerSlot(numActiveValidatorRegistry uint64) uint64 {
 	}
 	return boundOnValidatorRegistry
 }
+
+// AttestationParticipants returns the attesting participants indices.
+//
+// Spec pseudocode definition:
+//   def get_attestation_participants(state: BeaconState,
+//     attestation_data: AttestationData,
+//     participation_bitfield: bytes) -> List[int]:
+//     """
+//     Returns the participant indices at for the ``attestation_data`` and ``participation_bitfield``.
+//     """
+//
+//     # Find the relevant committee
+//     shard_committees = get_shard_committees_at_slot(state, attestation_data.slot)
+//     shard_committee = [x for x in shard_committees if x.shard == attestation_data.shard][0]
+//     assert len(participation_bitfield) == ceil_div8(len(shard_committee.committee))
+//
+//     # Find the participating attesters in the committee
+//     participants = []
+//     for i, validator_index in enumerate(shard_committee.committee):
+//         participation_bit = (participation_bitfield[i//8] >> (7 - (i % 8))) % 2
+//         if participation_bit == 1:
+//             participants.append(validator_index)
+//     return participants
+func AttestationParticipants(
+	state *pb.BeaconState,
+	attestationData *pb.AttestationData,
+	participationBitfield []byte) ([]uint32, error) {
+
+	// Find the relevant committee.
+	shardCommittees, err := ShardAndCommitteesAtSlot(state, attestationData.Slot)
+	if err != nil {
+		return nil, err
+	}
+
+	var participants *pb.ShardAndCommittee
+	for _, committee := range shardCommittees.ArrayShardAndCommittee {
+		if committee.Shard == attestationData.Shard {
+			participants = committee
+			break
+		}
+	}
+
+	if float64(len(participationBitfield)) != math.Ceil(float64(len(participants.Committee))) {
```

Comment:
im worried using floats will eventually lead to nasty errors - @rawfalafel expressed similar concerns in some past PRs especially if we're doing them for comparison. We should find an alternative to this.